client/webserver/api.go

```diff
@@ -681,12 +683,17 @@ func (s *WebServer) apiUser(w http.ResponseWriter, r *http.Request) {
 
 // writeAPIError logs the formatted error and sends a standardResponse with the
 // error message.
-func (s *WebServer) writeAPIError(w http.ResponseWriter, format string, a ...interface{}) {
-	errMsg := fmt.Sprintf(format, a...)
-	log.Error(errMsg)
+func (s *WebServer) writeAPIError(w http.ResponseWriter, err error) {
+	var cErr *core.Error
+	var code int
+	if errors.As(err, &cErr) {
+		code = cErr.Code()
+	}
 	resp := &standardResponse{
-		OK:  false,
-		Msg: errMsg,
+		OK:   false,
+		Msg:  err.Error(),
+		Code: &code,
```

Comment:
Um so I made an incomplete suggestion.  When the error is not detected as a core.Error above, it needs to be a nil pointer to omit the code.

```
var code *int
if errors.As(err, &cErr) {
    // set code
....
Code: code,
```

Reply:
should be good now